### PR TITLE
fix: Add test for migration

### PR DESCRIPTION
**Pull request by StanislavBreadless**
## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Comment by kelemeno** [comment]
To merge I want to wait for:  
https://github.com/matter-labs/era-contracts/pull/390
so that we can bump the contracts, and that so that CI passes.

File: core/lib/multivm/src/versions/vm_latest/tests/migration.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        tests::{
+            tester::{get_empty_storage, DeployContractsTx, TxType, VmTesterBuilder},
+            utils::read_test_contract,
+        },
+        HistoryEnabled,
+    },
+};
+
+#[test]
+fn test_migration_for_system_context_aa_interaction() {
```
Comment (mm-zk) [inline core/lib/multivm/src/versions/vm_latest/tests/migration.rs:15]: please add short description of what this test covers

File: core/lib/multivm/src/versions/vm_latest/tests/migration.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    // Now, we will just proceed with standard transaction execution.
+    // The bootloader should be able to update system context regardless of whether
+    // the upgrade transaction is there or not.
+    let account = &mut vm.rich_accounts[0];
+    let counter = read_test_contract();
+    let DeployContractsTx { tx, .. } = account.get_deploy_tx(&counter, None, TxType::L2);
+
+    vm.vm.push_transaction(tx);
+    let result = vm.vm.execute(VmExecutionMode::OneTx);
+    assert!(!result.result.is_failed(), "Transaction wasn't successful");
```
Comment (mm-zk) [inline core/lib/multivm/src/versions/vm_latest/tests/migration.rs:37]: maybe print the transaction result inside the error message?

File: core/lib/multivm/src/versions/vm_latest/tests/migration.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    let batch_result = vm.vm.execute(VmExecutionMode::Batch);
+    assert!(
+        !batch_result.result.is_failed(),
+        "Batch transaction wasn't successful"
```
Comment (mm-zk) [inline core/lib/multivm/src/versions/vm_latest/tests/migration.rs:42]: same here

**Review by mm-zk — APPROVED** [review]

**Pull request merged**